af/oometa/013-ooexplr-meta.install.sql

```diff
@@ -0,0 +1,81 @@
+BEGIN;
+
+select _v.register_patch( '013-ooexplr-meta', ARRAY[ '012-sha256-input-uniq' ], NULL );
+
+CREATE TABLE ooexpl_recent_msmt_count AS
+SELECT
+COUNT(msm_no) as msmt_count,
+probe_cc,
+probe_asn,
+test_name,
+test_start_time,
+bucket_date
+FROM measurement
+JOIN report ON report.report_no = measurement.report_no
+JOIN autoclaved ON autoclaved.autoclaved_no = report.autoclaved_no
+WHERE test_start_time > current_date - interval '31 day'
+GROUP BY probe_cc, probe_asn, test_name, test_start_time, bucket_date;
```

Comment:
The datums on this table will eventually be grouped on a per day basis. I think it's fine to also just consider measurement_start_time and not do this per report file aggregation, but it seemed like an easy optimisation at no cost of compromising the usage of data.

The problem with doing instead the aggregation on, say, a daily basis is that we then may have to recompute some stats or aggregation as new data from a bucket is processed.

Reply:
> recompute some stats or aggregation as new data from a bucket is processed

I assumed that the goal of that intermediate representation and partial pre-aggregation is to speed-up reading by creating a cube of stats and making reading of whole table unnecessary when the only thing that is needed is the count of rows.

That's exactly "doing some more writes to speed-up reads".

That makes me think that optimizing for the data representation that is compact and reflects API performance needs is the goal. Counting rows in a reprocessed bucket after incremental reprocessing is a relatively cheap operation: it computes stats over the rows that are likely in the page cache and re-writes stats for the single bucket. Basically, that's poor-man partial update for materialized view :)

Reply:
Yeah you raise a good point here. It may be worthwhile as soon as we have consolidated all the stats we need to drop this table and consolidate it into one or more fact tables that are updated using the bucket_date key.

At least that is the principle or thought I had with creating, for example, the `ooexpl_bucket_msmt_count` table, which is as you say basically a poor-man partial update for mat view with bucket_date as shard key.

Reply:
As suggested by @darkk we should probably group by truncated `measurement_start_time` using `date_trunc('day')`.